common/src/main/java/org/apache/celeborn/common/network/sasl/SaslUtils.java

```diff
@@ -66,4 +70,8 @@ static char[] encodePassword(String password) {
         .encodeToString(password.getBytes(StandardCharsets.UTF_8))
         .toCharArray();
   }
+
+  static void initializeSaslProviders() {
+    AnonymousSaslProvider.initializeIfNeeded();
+  }
```

Comment:
Instead of explicitly initializing, do we want to always initialize in a static block (essentially move contents of the `if` block in `initializeIfNeeded` to the static block to always add) ?

Reply:
Yes, I was trying that but the `SalUtils` class doesn't get loaded even though we are referencing the DEFAULT_REALM constant. This is because the compiler inlines the reference to the String constants. This causes the saslClient and the saslServer to be null. I think instead of adding some other way to load the SaslUtils class, it's better to explicitly load the SaslProviders.

Reply:
Yeah, compile time constants are inlined.
We have couple of options:
a) Make the constant a method returning the constant - looks a bit ugly.
b) Have a base (something lik `CelebornSaslBase` or `CelebornSaslSupport`) as the parent class - where all cross cutting client + server sasl code is hosted.

Both dont looks very attractive option, given this - let us drop this.
We can evolve it in future as required.